### DISCUSSION TITLE
post/trimming-wav-file-using-naudio

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#18] [Announcements]
# post/trimming-wav-file-using-naudio

Mark Heath's Development Blog

https://markheath.net/post/trimming-wav-file-using-naudio

**Comment by gigjacker** [comment]
Hi, and thank you for posting this example :-) 

Looks like you have a typo in TrimWavFile on this line:

      while (reader.Position < endpos)  // missing uppercase P as the parameter name is endPos

(Would make it easier to ask questions if you provided line numbers... :-) ) 

Thanks again.

**Reply by markheath** [reply]
thanks, will get it fixed

**Comment by nanimuwu** [comment]
There's an issue when I try to do this on many (2k+) audio files with a very small time frame between (often couple hundred milliseconds). In small numbers it's fine, but occasionally it will create a wav file of size 0. As I am a novice programmer, I do not know how to troubleshoot this. Is it possible to do with the Godot game engine I'm using this with tangentially, or is it just a quirk of the library?

**Reply by nanimuwu** [reply]
This might be an issue with the concatenationsampler i used to restitch the files for testing, maybe disregard the comment? sorry about that
